File: 프로그래머스/0/181854. 배열의 길이에 따라 다른 연산하기/배열의 길이에 따라 다른 연산하기.py
def solution(arr, n):
    answer = []
    if len(arr) % 2 == 0:
        eo = 1
        for i in arr:
            if eo == 1:
                answer.append(i)
                eo = 2
            else:
                answer.append(i+n)
                eo = 1
                
    else:
        eo = 1
        for i in arr:
            if eo == 1:
                answer.append(i+n)
                eo = 2
            else:
                answer.append(i)
                eo = 1
    return answer

File: 프로그래머스/0/181854. 배열의 길이에 따라 다른 연산하기/배열의 길이에 따라 다른 연산하기.js
function solution(arr, n) {
    var answer = [];
    if (arr.length%2) {
        for (let [i, v] of arr.entries()) {
            answer.push(i%2 ? v : v+n);
        }
    } else {
        for (let [i, v] of arr.entries()) {
            answer.push(i%2?v+n:v)
        }
    }
    return answer;
}
